### PR TITLE
FIX: fix bad string handling in type1font

### LINKED ISSUE #4470: Issue encoding type1 fonts with python 3
Initially reported at https://github.com/ContinuumIO/anaconda-issues/issues/337

```python
import numpy as np
import matplotlib as mpl
import matplotlib.pylab as plt

mpl.rcdefaults()
# This doesn't work
 mpl.rcParams['font.serif'] = "Times, serif"
# This does
#mpl.rcParams['font.serif'] = "Bitstream Vera Serif, serif"
mpl.rcParams['text.usetex'] = "true"

x = np.random.normal(size=100)
y = np.random.normal(size=100)
plt.scatter(x, y)
plt.xlabel("$\gamma$")
plt.savefig("foo.png")
print("foo.png saved")
plt.savefig("bar.pdf")
print("bar.pdf saved")
```

traceback: 

```python

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-3a778cad97aa> in <module>()
----> 1 exec(compile(open('/tmp/ipython-56520H0.py').read(), '/tmp/ipython-56520H0.py', 'exec')) # PYTHON-MODE

/tmp/ipython-56520H0.py in <module>()

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/pyplot.py in savefig(*args, **kwargs)
    577 def savefig(*args, **kwargs):
    578     fig = gcf()
--> 579     res = fig.savefig(*args, **kwargs)
    580     draw()   # need this if 'transparent=True' to reset colors
    581     return res

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/figure.py in savefig(self, *args, **kwargs)
   1507             self.set_frameon(frameon)
   1508 
-> 1509         self.canvas.print_figure(*args, **kwargs)
   1510 
   1511         if frameon:

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backends/backend_qt5agg.py in print_figure(self, *args, **kwargs)
    158 
    159     def print_figure(self, *args, **kwargs):
--> 160         FigureCanvasAgg.print_figure(self, *args, **kwargs)
    161         self.draw()
    162 

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, **kwargs)
   2223                 orientation=orientation,
   2224                 bbox_inches_restore=_bbox_inches_restore,
-> 2225                 **kwargs)
   2226         finally:
   2227             if bbox_inches and restore_bbox:

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backends/backend_pdf.py in print_pdf(self, filename, **kwargs)
   2513                 file.endStream()
   2514             else:            # we opened the file above; now finish it off
-> 2515                 file.close()
   2516 
   2517 

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backends/backend_pdf.py in close(self)
    552         self.endStream()
    553         # Write out the various deferred objects
--> 554         self.writeFonts()
    555         self.writeObject(self.alphaStateObject,
    556                          dict([(val[0], val[1])

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backends/backend_pdf.py in writeFonts(self)
    648                 matplotlib.verbose.report('Writing Type-1 font', 'debug')
    649                 fonts[Fx] = self.embedTeXFont(filename,
--> 650                                               self.dviFontInfo[filename])
    651             else:
    652                 # a normal TrueType font

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/backends/backend_pdf.py in embedTeXFont(self, texname, fontinfo)
    713         t1font = type1font.Type1Font(fontinfo.fontfile)
    714         if fontinfo.effects:
--> 715             t1font = t1font.transform(fontinfo.effects)
    716         fontdict['BaseFont'] = Name(t1font.prop['FontName'])
    717 

/home/tcaswell/source/my_source/matplotlib/lib/matplotlib/type1font.py in transform(self, effects)
    334                 if six.PY3 and isinstance(value, int):
    335                     value = chr(value)
--> 336                 value = value.encode('latin-1')
    337                 buffer.write(value)
    338             result = buffer.getvalue()

AttributeError: 'bytes' object has no attribute 'encode'

```

### PR DESCRIPTION
**Pull request by tacaswell**
Do not try to encode a bytestring in python 3.

closes #4470

### CODE REVIEW

**Comment by tacaswell** [comment]
Also closes ContinuumIO/anaconda-issues#337

**Comment by tacaswell** [comment]
The random test failures are from running too many processes on travis and latex stepping on it's own toes.

File: lib/matplotlib/tests/test_backend_pdf.py [hunk 1]
```diff
@@ -87,3 +87,13 @@ def test_multipage_keep_empty():
             pdf.savefig()
         assert os.path.exists(filename)
     os.remove(filename)
+
+
+@image_comparison(baseline_images=['test_type1font_encoding'],
+                  extensions=['pdf'])
+def test_type1font_encoding():
+    # This doesn't work
```
Comment (efiring) [inline lib/matplotlib/tests/test_backend_pdf.py]: Did you intend to leave this comment here?  It's baffling.

File: .travis.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ matrix:
 
 install:
   - pip install -q --use-mirrors nose python-dateutil $NUMPY pep8==1.5.7 pyparsing pillow
-  - sudo apt-get update && sudo apt-get -qq install inkscape libav-tools mencoder
+  - sudo apt-get update && sudo apt-get -qq install inkscape libav-tools mencoder dvipng texlive-latex-base texlive-latex-extra texlive-fonts-recommended
```
Comment (efiring) [inline .travis.yml]: Perhaps there's no alternative, but I think all those tex packages are huge and will slow down the builds. The existing lines below indicate that the penalty might be reduced by using --no-install-recommends.

Comment (jenshnielsen) [inline .travis.yml]: I did that --no-install-recommends for building the documentation. As I remember it it saved a few 100 MB download. In any case this is only relevant on the colour overhaul branch. On master we are using a container build which handles this different.

File: lib/matplotlib/tests/test_backend_pdf.py (hunk 1 above)
Comment (tacaswell) [inline lib/matplotlib/tests/test_backend_pdf.py]: That is a typo from modifying the bug report example to be a test, should have been removed.

**Comment by tacaswell** [comment]
Closing in favor of #4522

**Pull request closed without merging**